src/sentry/auth/idpmigration.py

```diff
@@ -67,5 +71,11 @@ def verify_new_identity(user: User, org: Organization, key: str) -> bool:
     :param key: the one-time verification key
     :return: whether the key is valid
     """
-    # user cluster.hgetall(key) to get from redis
-    raise NotImplementedError  # TODO
+    cluster = redis.clusters.get("default").get_local_client_for_key(_REDIS_KEY)
+
+    verification_key = f"auth:one-time-key:{key}"
+    verification_value_byte = cluster.hgetall(verification_key)
+    if not verification_value_byte:
+        return ""
+
+    return verification_key
```

Comment:
I'd rather return a boolean than assign meaning to an empty string. It looks like the intent here is to have this function own the Redis prefix, and then `IDPEmailVerificationView` uses the returned value?

It might be better to have a stand-alone function that transforms the bare key into a Redis key:
```python
def get_redis_key(verification_key: str) -> str:
    return f"auth:one-time-key:{verification_key}"
```
which could be called both from here and from `IDPEmailVerificationView`. Then `verify_new_identity` would only need to return `bool(verification_value_byte)`.

Reply:
ah this makes sense, i thought i needed to get the key from verify_new_identity but i completely forgot i already had it in IDPEmailVerificationView